controller/grpc.go

```diff
@@ -150,6 +150,26 @@ func (ss *ScanService) ScannerRegister(ctx context.Context, data *share.ScannerR
 	}
 }
 
+// HealthCheck checks if the scanner is in the list of controller Consul key-value pairs and if the controller is alive
+// This function follows the logic from how scannerRegister function puts the scanner into the KV store to retrieve it.
+func (ss *ScanService) HealthCheck(ctx context.Context, data *share.ScannerRegisterData) (*share.ScannerAvailable, error) {
+	visible := false
```

Comment:
optional: `found` would be more appropriate word choice than `visible` as you need a UI to really see it.